### PR TITLE
Added don information!

### PR DESCRIPTION
**Pull request by scholvat**
Added a way to let grebelites see which dons (and campus hosts) are on, on late supper, how to contact them, and where they are located. This is related to skybunk-server PR https://github.com/CGUC/skybunk-server/pull/70.

Features include:
-A new page in the settings menu to view don information
-A UI for dons to change their status or change the status of other dons, change location, and set clockout time
-Added phone number to memberEdit and user profiles. This is mainly for dons so people can contact them, but others can add it too if they want
-Changed isAdmin to role, so that we can specify who the user is. This can be useful in the future for special permissions for staff members, alumni, etc.

Related issues:  #93 #42 #31 

Screenshots!

[A don's view when they are on](https://user-images.githubusercontent.com/10137206/48928890-5eef9780-ee98-11e8-94fe-b00c7b23db19.png)

[A don's view when they are off](https://user-images.githubusercontent.com/10137206/48928891-5eef9780-ee98-11e8-9db1-f7dc4060d76d.png)

[A normal Grebelite's view of the don info](https://user-images.githubusercontent.com/10137206/48928892-5eef9780-ee98-11e8-8f49-5d108517b04b.png)

[Datetime editor for changing clockout time](https://user-images.githubusercontent.com/10137206/48928976-200e1180-ee99-11e8-814d-bbadbdaa5c95.png)

### CODE REVIEW

**Comment by picklechips** [comment]
a) Just want to say this is sick, good work dude

b) I have like 5 things due on Monday, so I probably won't be able to review until Tuesday (Since there's some significant changes, I'd like to look over it somewhat thoroughly)

**Comment by neilbrub** [comment]
This looks great. Have yet to go through it with a fine-tooth comb, but the mechanics look very intuitive and I like the defaults, I think they will really add to usability.
The one question I have is whether dons will want to keep on top of setting their status every time their shift comes around - for some it's already a burden to keep a door sign updated! And if they don't, this feature will lose its credibility and won't be super effective :( If we could take advantage of the agenda even more so that dons could set their on/off days (usually fairly constant) and have that as the default that would be another huge step. However I don't think it would hurt to try this out as a feature-under-test for a bit and see how it goes.
Will try to make some time in the next few days to actually play around with it a bit :)

**Comment by scholvat** [comment]
@neilbrub Some of my logic for my design:
1) Full schedules sounded like a lot of work to start out with, so I started out a little less ambitious.
2) I found it difficult to design a mobile UI that lets the don control their full schedule and the UI not get cluttered.
3) My idea was to continue to let dons organise their schedules however they want, and just use skybunk as a communication tool. Since each don team organizes their schedules a little differently, I didn't want to restrict that within the Skybunk app.
4) I'm thinking of also implementing a Wifi positioning system (WPS) so that the don locations are automatically updated as the dons move around Grebel. Hopefully this can be done by mapping WAP to locations around Grebel, but there are a ton of technical limitations that I'll have to test out when I get back to Waterloo. This should help keep the app up to date, and dons can manually update their location if they feel it isn't their style.
5) Let me dream even further, but you could digitise the hallway signs with some cheap tablets that are running skybunk, so the dons only have to update their information in one place. This is unlikely to actually work out, but it would be interesting!

TLDR: doing a full don schedule was too much work, so I started with this. If that extra work means that the tool will actually be used, then its worth the work.

If you are interested in testing this out yourself, I do have a dev server setup at "http://ec2-18-216-251-6.us-east-2.compute.amazonaws.com:3000" with some dummy information. Everyone's password is "password" so you can login as yourself or as a don.

File: views/Comments/Comments.jsx [hunk 1]
```diff
@@ -250,7 +250,7 @@ export default class CommentsView extends React.Component {
                         data={comment}
                         updateComment={this.updateResource}
                         enableEditing={enableCommentEditing}
-                        enableDeleting={loggedInUser.isAdmin}
+                        enableDeleting={loggedInUser.role.includes("admin")}
```
Comment (neilbrub) [inline views/Comments/Comments.jsx]: Here and line 237 we should add some error checking to ensure user objects actually have the `role` field (don't think existing objects get updated with the model in the twin server PR). If the object doesn't have the field `enableDeleting` should be false.

Comment (picklechips) [reply views/Comments/Comments.jsx]: Might be worth making the role field on the model required (with a default value of empty string), then writing a script to go and update all the past documents -- Just to avoid having to check for the field everywhere we use it.

**Comment by picklechips** [comment]
On a somewhat unrelated note, 

> I do have a dev server setup at "http://ec2-18-216-251-6.us-east-2.compute.amazonaws.com:3000" with some dummy information

I've been meaning to add something to the readme about this, but using [ngrok](https://ngrok.com/) for testing out a dev-server is probably easier than hosting it on aws.  Basically you just run your server locally then run `./ngrok http <PORT-SERVER-IS-RUNNING-ON>` and it will temporarily host your local server & give you a link you can use in the mobile app.

File: views/DonInfo/DonInfo.jsx [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    this.props.navigation.setParams({
+      save: this.onSave,
+      saveState: 'disabled'
+    });
+  }
+
+  filterDons(users){
+    var dons = []
+    users.forEach(user =>{
+      if((user.role && user.role.includes("don"))==1){
```
Comment (picklechips) [inline views/DonInfo/DonInfo.jsx]: Not sure if it matters but why the `==1`

**Comment by picklechips** [comment]
A couple minor questions, but otherwise I I don't have any big issues with this - pending the server changes. Looks good 👍

**Comment by scholvat** [comment]
In regards to the ngrok, I was using aws so others could demo the app if they wanted (I was going to send a preprod version of the app to the upcoming dons but no one took me up on the offer). Since my computer isn't always on, I just put it on my free AWS server. But ngrok is cool, I didn't know that existed :)

**Comment by scholvat** [comment]
I assume ngrok gets through eduroam's security issues?

File: views/DonInfo/DonInfo.jsx (hunk 2 above)
Comment (scholvat) [reply views/DonInfo/DonInfo.jsx]: Because that's the C style guide that I'm used to... will fix :)

**Comment by neilbrub** [comment]
LGTM once server is in.
